contracts/ERC777PresetPausablePermissioned.sol

```diff
@@ -23,6 +23,65 @@ contract ERC777PresetPausablePermissioned is
    */
   uint256[50] private __gap;
 
+  /**
+   * @dev An event emitted when a batch of transfers are bundled into a single transaction
+   */
+  event SentBatch(
+    address indexed from,
+    address[] recipients,
+    uint256[] amounts,
+    bytes[] userData,
+    bytes[] operatorData,
+    bool[] requireReceptionAck
+  );
+
+  /**
+   * @notice Batches multiple transfers into a single transaction
+   *
+   * @param recipients address[] list of recipient addresses
+   * @param amounts uint256[] list of amounts to transfer
+   * @param userData bytes[] list of extra information provided by the token holder (if any)
+   * @param operatorData bytes[] list of extra information provided by the operator (if any)
+   * @param requireReceptionAck list of requirements (if true, contract recipients are required to implement
+   * ERC777TokensRecipient)
+   *
+   * Emits a {SendBatch} event.
+   *
+   * ##### Requirements:
+   *
+   * - the contract must not be paused
+   */
+  function batchSend(
```

Comment:
Is this copied over from the OZ implementation or is this net new?  Do we want to set a limit on the batch size?  I assume otherwise we're eventually going to run into a size limit here beyond which the batch fails?

Reply:
No, this is custom. Size limit is going to depend on more than just number of recipients since send can call an external contract. We'd need to set a limit some other way if we want to explore that